packages/tupaia-web-server/src/routes/MapOverlaysRoute.ts

```diff
@@ -133,21 +129,29 @@ export class MapOverlaysRoute extends Route<MapOverlaysRequest> {
               code: overlay.code,
               reportCode: overlay.report_code,
               legacy: overlay.legacy,
+              sortOrder: relation.sort_order,
```

Comment:
For the sake of the two layered sort, it's easiest if we have access to both `sort_order` and `name` on a single object. But usually `sort_order` is attached to the `relation` entry, while `name` is attached to the `map_overlay[_group]`

So we just temporarily attach `sortOrder` up until we sort the children below, then we discard it so it's not returned to the frontend